common/hyperscaler/rules/model/config.go

```diff
@@ -0,0 +1,26 @@
+package model
+
+import (
+	"fmt"
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+type RulesConfig struct {
+	Rules []string `yaml:"rule"`
+}
+
+func (c *RulesConfig) Load(file string) (*RulesConfig, error) {
```

Comment:
This is file name of type string. Not file.
But nevertheless I would consider Load using not file name as input, but either interface or []byte collection like in yaml.Unmarshal - to make it easier to test.

Reply:
Changed `file` to `filePath`. I do not want the method to get a byte array because it would just wrap Unmarshal invocation which is excessive.

Reply:
And you just wrap two things instead of one.
Os.Read and yaml.Unmarshal.
With such approach you cannot test this little part of logic without having a file.

Reply:
I do not want to mirror libraries' logic by creating a method for every function. It is unnecessary here.